### ISSUE TITLE
Zesty support/build for spell checker

### ISSUE THREAD

**Issue by mcg** [raelgc/scudcloud#571]
## ScudCloud Version
```
scudcloud --version
ScudCloud 1.58
Python 3.5.3
Qt 5.7.1
PyQt 5.7
SIP 4.18.1
```

## Distro and Desktop info

  - Distribution: Ubuntu
  - Version/Release: 17.04
  - Desktop environment: Ubuntu GNOME

## Expected behavior

Working spellcheck.

## Actual behavior
```
The following packages have unmet dependencies:
 libqtwebkit-qupzillaplugins : Depends: libhunspell-1.3-0 (>= 1.3.3) but it is not installable
```

I am using the yakkety PPA in zesty. Works except for spellcheck due to that hunspell version issue. Any chance we can get a Zesty build in the PPA?

**Comment by raelgc** [comment]
@mcg, I just published both packages (Scudcloud and libqtwebkit-qupzillaplugins) for Zesty. Sorry about miss Zesty.

**Comment by mcg** [comment]
That was quick! Unfortunately, using the zesty repo. the same libhunspell dependency issue happens.

**Comment by raelgc** [comment]
According the [package page](http://packages.ubuntu.com/en/zesty/libhunspell-1.4-0), the package name is now `libhunspell-1.4-0`.

I'll build a fix for it.

**Comment by raelgc** [comment]
I tried to change the requirements to accept `libhunspell` `1.3` or `1.4`. But the [build failed](https://launchpadlibrarian.net/317326648/buildlog_ubuntu-zesty-amd64.libqtwebkit-qupzillaplugins_1.2-2_BUILDING.txt.gz). It appears it'll require to generate the Qt build again in Zesty.

For this I'll require a Zesty VM. I'll require help or wait some days, until my Internet is properly installed (I'm working under a 3G connection).

**Comment by mcg** [comment]
Does Launchpad no longer provide PPA build services? It's been years since I used them.

**Comment by UnitedMarsupials-zz** [comment]
How do I trigger spell-checking of my messages? I have hunspell installed. I traced the scudcloud process and see, that it finds the hunspell-executable -- but there are no obvious attempts to use it and the misspelled message go out into the world without any attempt to warn me and/or correct them. Is there some setting I need to turn on, or is this functionality simply not implemented yet?

**Comment by raelgc** [comment]
@UnitedMarsupials are you on which distro and version? For now, it works only on Ubuntu until 16.10. You need to build `libqtwebkit-qupzillaplugins` for you distro and version. It's a webkit plugin (not a webbrowser plugin) to enable spell checking.

**Comment by UnitedMarsupials-zz** [comment]
@raelgc, as we already established elsewhere, I'm a FreeBSD-user -- the term "distro" does not exist in our world :)

> For now, it works only on Ubuntu until 16.10.

Could you enumerate, what software -- libraries, executables -- are required for it to work? Is it more than just finding the `hunspell` executable, for example -- which scudcloud does find, according to process-trace?..

Lastly, is it supposed to "just work", when everything is in place, or do I need to request it though the GUI somewhere? Thanks!

**Comment by raelgc** [comment]
Sorry @UnitedMarsupials.

You need to build [this library](https://github.com/raelgc/libqtwebkit-qupzillaplugins): it contains 2 webkit plugins (it's different than navigator plugins, they extend the webkit funcionality).

After build it and install, it'll just work if hunspell related dictionary is installed (of course, you need to restart scudcloud).

**Comment by UnitedMarsupials-zz** [comment]
> You need to build [this library](/raelgc/libqtwebkit-qupzillaplugins)

Ok... And that's a distinct piece of software from [QupZilla/qupzilla-plugins](/QupZilla/qupzilla-plugins)? Is it your own project of general usefulness, or something only useful for scudcloud-users? I'm asking to better decide, whether to create a separate port for it, or to bundle it with scudcloud...

**Comment by raelgc** [comment]
> that's a distinct piece of software from QupZilla/qupzilla-plugins?

No, it's just a `.deb` wrapper for [QupZilla/qupzilla-plugins](/QupZilla/qupzilla-plugins).

**Comment by UnitedMarsupials-zz** [comment]
> No, it's just a .deb wrapper for

Then a FreeBSD-user does not need to build it, do I? :-) I will create a FreeBSD port of `qupzilla-plugins`...

But, meanwhile, why is scudcloud even looking for hunspell, if the plugins aren't available?
